src/Umbraco.Web/Editors/CodeFileController.cs

```diff
@@ -0,0 +1,155 @@
+﻿using AutoMapper;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web.Http;
+using Umbraco.Core.Models;
+using Umbraco.Web.Models.ContentEditing;
+using Umbraco.Web.Mvc;
+using Umbraco.Web.WebApi;
+
+namespace Umbraco.Web.Editors
+{
+    [PluginController("UmbracoApi")]
+    public class CodeFileController : UmbracoAuthorizedJsonController
```

Comment:
Ye, that's not ideal. How would you go about doing it for this one where each method could relate to three different trees? 

For now I've just added the UmbracoApplicationAuthorizeAttribute for settings (which again isn't ideal because if you have access to developer but not settings you it wouldn't work - but that's a crazy edge case until new org of trees is in place and I'm fine with that)